internal/dataplane/kong_client.go

```diff
@@ -121,6 +125,7 @@ func NewKongClient(
 ) (*KongClient, error) {
 	// build the client object
 	cache := store.NewCacheStores()
+	indexers := store.NewCacheIndexers()
```

Comment:
I'm wondering whether we have to put the indexers into the `dataplane.KongClient` and extend its interface unnecessarily. As I can see it already has a very wide interface and I've got a gut feeling that we're treating it as a good place for any controller dependency while we shouldn't as it's originally meant to be the interface between controllers and the translation stage. As I understand, the translation stage doesn't need any of the newly introduced interface methods (like `SetObjectReference`, `DeleteObjectIfNotReferred` etc.). 

WDYT about implementing this part of the interface as a separate type under `controllers/reference` and injecting it only to the controllers that depend on it?

Reply:
Thanks. It is a great suggestion, will do it.

Reply:
refactored the code to move definition of indexer to `controller/reference` package, and removed the field in dataplane client.